lib/components/narrative/narrative-itineraries.js

```diff
@@ -38,19 +70,87 @@ class NarrativeItineraries extends Component {
   render () {
     const {
       activeItinerary,
+      activeSearch,
+      filter,
       itineraries,
       itineraryClass,
+      pending,
+      persistence,
       realtimeEffects,
+      sort,
       useRealtime
     } = this.props
-    if (!itineraries) return null
-
+    if (!activeSearch) return null
+    const itineraryIsExpanded = activeItinerary !== undefined && activeItinerary !== null && this.state.showDetails
+    // Construct loading divs as placeholders while all itineraries load.
+    const loadingItineraryCount = this.props.modes.combinations
+      ? this.props.modes.combinations.length - itineraries.length
+      : 0
+    const loadingItineraryDivs = Array.from(
+      {length: loadingItineraryCount},
+      (v, i) =>
+        <div key={i} className='option default-itin'>
+          <SkeletonTheme color='#ddd' highlightColor='#eee'>
+            <Skeleton count={3} />
+          </SkeletonTheme>
+        </div>
+    )
     return (
       <div className='options itinerary'>
-        <div className='header'>
-          We found {itineraries.length} itineraries:
+        <div className='header' style={{display: 'flex', justifyContent: 'space-between'}}>
+          {itineraryIsExpanded
+            ? <>
+              <button
+                className='clear-button-formatting'
+                onClick={this._toggleDetailedItinerary}>
+                <i className='fa fa-arrow-left' /> View all options
+              </button>
+              {/* FIXME: only save if meets requirements (e.g., is transit + non-realtime mode) */}
+              {persistence && persistence.enabled
+                ? <button
+                  className='clear-button-formatting'
+                  onClick={this._saveTrip}>
+                  <i className='fa fa-plus-circle' /> Save trip
+                </button>
+                : null
+              }
+            </>
+            : <>
+              <div>
+                {pending
+                  ? 'Finding your options...'
+                  : <span>We found {itineraries.length} itineraries.</span>
```

Comment:
On screens that are less than full-width, there are some layout issues:
![image](https://user-images.githubusercontent.com/56846598/88731301-0d83bb00-d105-11ea-8cfb-7b1ec7f2fc45.png)

Reply:
I do love the layout on full-width, though!

Reply:
Thanks! Yes, we need to resolve mobile layouts as well. I was kind of holding off on spending too much time on the layout concerns until we addressed mobile stuff more fully.